tools/RAiDER/models/weatherModel.py

```diff
@@ -581,73 +582,203 @@ def _checkForNans(self):
         self._t = fillna3D(self._t)
         self._e = fillna3D(self._e)
 
-    def write2HDF5(self, outName=None):
+    def write2NETCDF4(self, outName=None, NoDataValue=-3.4028234e+38, chunk=128):
```

Comment:
Here's where I think a generic "write" method with a file type parameter would be better. Suggestion: 
```suggestion
    def write(self, outName=None, fmt='netcdf4'):
        if fmt == 'netcdf4':
            # do your new code
        elif fmt == 'hdf5':
            # do the old code
        else:
            raise ValueError('fmt {} not a valid fmt for writing the weather model results')
            # (or we could simply change the fmt to the default value)
```